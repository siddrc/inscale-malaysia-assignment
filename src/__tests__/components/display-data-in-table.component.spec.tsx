import React from 'react';
import {
    render,
    cleanup
  } from '@testing-library/react';
import {DisplayDataInTable,DisplayDataInTableProps} from '../../components/display-data-in-table.component'
  
afterEach(()=>{
    cleanup();
});
describe('Display data in table component unit test cases',()=>{
    it('for no data is should display no data found test-id',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[],
            headers:[],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getByTestId} = component;
        expect(getByTestId("no-data-found")).toBeTruthy();
    });
    it('for no data is should display no data found test-id-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[],
            headers:[],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    });
    it('for given header data, it should display those many header cells',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("header-cells").length).toEqual(2);
    });
    it('for given header data, it should display those many header cells-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    });
    it('for given header data, it should display the correct header',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("header-cells")[0].textContent).toEqual("Header1");
    });
    it('for given header data, it should display the correct header-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    });
    it('for a given data, it will not display data if no keys are present',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("no-keys-defined").length).toEqual(displayDataInTableProps.data.length)
    })
    it('for a given data, it will not display data if no keys are present-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:[]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    })
    it('for a given data, it must display data',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("data-cells").length).toEqual(1);
    })
    it('for a given data, it must display data-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    })
    it('for a given data, it will not display, if keys are present by incorrect',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"},{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key1"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("incorrect-keys-defined").length).toEqual(2)
    });
    it('for a given data, it will not display, if keys are present by incorrect-snapshot test',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"},{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key1"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    });
    it('for a given data, it display the correctly as per the snapshot',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"},{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        expect(component).toMatchSnapshot()
    })
    it('for a given data in correct conditions, the number of rows must match length of data array',()=>{
        const displayDataInTableProps:DisplayDataInTableProps = {
            data:[{"key":"value"},{"key":"value"}],
            headers:["Header1","Header2"],
            keys:["key"]
        };
        const component = render(<DisplayDataInTable {...displayDataInTableProps}/>);
        const {getAllByTestId} = component;
        expect(getAllByTestId("all-data-rows").length).toBe(displayDataInTableProps.data.length);
    })
})