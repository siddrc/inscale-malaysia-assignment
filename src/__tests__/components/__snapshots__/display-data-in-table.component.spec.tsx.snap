// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display data in table component unit test cases for a given data, it display the correctly as per the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="data-cells"
            >
              value
            </td>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="data-cells"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="data-cells"
          >
            value
          </td>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="data-cells"
          >
            value
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for a given data, it must display data-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="data-cells"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="data-cells"
          >
            value
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for a given data, it will not display data if no keys are present-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="no-keys-defined"
            >
              No keys defined
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="no-keys-defined"
          >
            No keys defined
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for a given data, it will not display, if keys are present by incorrect-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="incorrect-keys-defined"
            />
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="incorrect-keys-defined"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="incorrect-keys-defined"
          />
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="incorrect-keys-defined"
          />
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for given header data, it should display the correct header-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="no-keys-defined"
            >
              No keys defined
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="no-keys-defined"
          >
            No keys defined
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for given header data, it should display those many header cells-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-striped"
      >
        <tbody>
          <tr>
            <th
              data-testid="header-cells"
            >
              Header1
            </th>
            <th
              data-testid="header-cells"
            >
              Header2
            </th>
          </tr>
          <tr
            data-testid="all-data-rows"
          >
            <td
              data-testid="no-keys-defined"
            >
              No keys defined
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-striped"
    >
      <tbody>
        <tr>
          <th
            data-testid="header-cells"
          >
            Header1
          </th>
          <th
            data-testid="header-cells"
          >
            Header2
          </th>
        </tr>
        <tr
          data-testid="all-data-rows"
        >
          <td
            data-testid="no-keys-defined"
          >
            No keys defined
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display data in table component unit test cases for no data is should display no data found test-id-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="no-data-found"
      >
        No data found
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="no-data-found"
    >
      No data found
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
